### PR TITLE
Update View and ViewData to have only one Aggregation/Data

### PR DESCRIPTION
**Pull request by songy23**
Per our discussion, now each `View` will have one `Aggregation` and each `ViewData` will have one type of `AggregationData`. Also update `MeanData` so that users can get count from it.

Might be easier to review by each commit.

`Distribution` Aggregation will be added in PR #630 .

### CODE REVIEW

File: core_impl/src/test/java/io/opencensus/implcore/stats/IntervalBucketTest.java [hunk 1]
```diff
@@ -49,28 +43,24 @@
   private static final Duration MINUTE = Duration.create(60, 0);
   private static final Duration NEGATIVE_TEN_SEC = Duration.create(-10, 0);
   private static final Timestamp START = Timestamp.create(60, 0);
-  private static final BucketBoundaries BUCKET_BOUNDARIES =
-      BucketBoundaries.create(Arrays.asList(-10.0, 0.0, 10.0));
-  private static final List<Aggregation> AGGREGATIONS_V1 =
-      Collections.unmodifiableList(Arrays.asList(
-          Sum.create(), Count.create(), Histogram.create(BUCKET_BOUNDARIES)));
+  private static final Mean MEAN = UnreleasedApiAccessor.createMean();
```
Comment (dinooliva) [inline core_impl/src/test/java/io/opencensus/implcore/stats/IntervalBucketTest.java]: createMean should be public now

Comment (songy23) [reply core_impl/src/test/java/io/opencensus/implcore/stats/IntervalBucketTest.java]: Correct, Mean is exposed in the next PR: https://github.com/census-instrumentation/opencensus-java/pull/630/files#diff-d4eaa57f3d0967756e7c291e24c53968R125, along with other changes on AggregationData.

File: core_impl/src/test/java/io/opencensus/implcore/stats/ViewManagerImplTest.java [hunk 2]
```diff
@@ -569,38 +543,14 @@ public void testMultipleViewsDifferentMeasures() {
...
         .isEqualTo(CumulativeData.create(Timestamp.create(2, 0), Timestamp.create(4, 0)));
     StatsTestUtil.assertAggregationMapEquals(
         viewData2.getAggregationMap(),
         ImmutableMap.of(
-            Arrays.asList(VALUE), StatsTestUtil.createAggregationData(AGGREGATIONS, 2.2)),
-        EPSILON);
-  }
-
-  @Test
-  public void testGetCumulativeViewDataWithoutBucketBoundaries() {
```
Comment (sebright) [inline core_impl/src/test/java/io/opencensus/implcore/stats/ViewManagerImplTest.java]: I think we should keep this test for when we add back Distribution.

File: core/src/test/java/io/opencensus/stats/RpcViewConstantsTest.java [hunk 3]
```diff
@@ -47,30 +44,6 @@ public void testConstants() {
         0.0, 1.0, 2.0, 4.0, 8.0, 16.0, 32.0, 64.0, 128.0, 256.0, 512.0, 1024.0, 2048.0,
         4096.0, 8192.0, 16384.0, 32768.0, 65536.0).inOrder();
 
-    // Test Aggregations
-    assertThat(RpcViewConstants.AGGREGATION_NO_HISTOGRAM)
-        .containsExactly(Sum.create(), Count.create())
-        .inOrder();
-    assertThat(RpcViewConstants.AGGREGATION_WITH_BYTES_HISTOGRAM)
```
[the remark this answers is not in the document]
Comment (songy23) [reply core/src/test/java/io/opencensus/stats/RpcViewConstantsTest.java]: These tests will be added in #630, after `RpcViewConstants` are updated to use the new `Distribution`.

File: core/src/test/java/io/opencensus/stats/ViewDataTest.java [hunk 4]
```diff
@@ -208,28 +198,14 @@ public void preventStartTimeLaterThanEndTime() {
   private static final BucketBoundaries BUCKET_BOUNDARIES = BucketBoundaries.create(
       Arrays.asList(10.0, 20.0, 30.0, 40.0));
   
-  private static final List<Aggregation> AGGREGATIONS = Collections.unmodifiableList(Arrays.asList(
-      Sum.create(), Count.create(), Range.create(), Histogram.create(BUCKET_BOUNDARIES),
-      Mean.create(), StdDev.create()));
+  private static final Mean MEAN = Mean.create();
 
-  private static final ImmutableMap<List<TagValueString>, List<AggregationData>> ENTRIES =
+  private static final ImmutableMap<List<TagValueString>, ? extends AggregationData> ENTRIES =
```
[the remark this answers is not in the document]
Comment (songy23) [reply core/src/test/java/io/opencensus/stats/ViewDataTest.java]: Done.

File: core_impl/src/test/java/io/opencensus/implcore/stats/ViewManagerImplTest.java (hunk 2 above)
Comment (songy23) [reply core_impl/src/test/java/io/opencensus/implcore/stats/ViewManagerImplTest.java]: Done, though currently `Distribution` needs to have non-null `BucketBoundaries`.

Comment (sebright) [reply core_impl/src/test/java/io/opencensus/implcore/stats/ViewManagerImplTest.java]: Sorry, my comment wasn't very clear.  I meant that I think we should test Distribution in all of the other tests and use this test to test a different aggregation (e.g., Mean) after you add Distribution.  I think that leaving it commented out for now is fine, though.

**Comment by songy23** [comment]
Thanks!

**Pull request merged**